### PR TITLE
[Complete] Rails 5.0.0 support 

### PR DESCRIPTION
**Pull request by Fudoshiki**
- [x] Update Travis CI setup, target on Mongoid master branch
- [x] Refactor code for Mongoid master branch
- [x] Fix test
- [x] Build pass

### CODE REVIEW

**Comment by Fudoshiki** [comment]
@simi  Complete!

**Comment by Fudoshiki** [comment]
@simi Please check PR, monogoid_paranoia is dep for mongoid-slug https://github.com/digitalplaywright/mongoid-slug/pull/219

**Comment by Fudoshiki** [comment]
@simi Can you look at the PR?

**Comment by johnnyshields** [comment]
as this breaks compat with Mongoid 4.x and 5.x we need to release this as a major version bump

**Comment by simi** [comment]
Thanks for this patch!

Finally I went thru the code.

Can I ask for few final changes?

1. remove everything unrelated to achieve Rails 5/Mongoid 6 compatibility (aka syntax change, .gitignore change)
2. instead of changing `belongs_to` option to `required: false`, can you change global config in spec helper `Mongoid.belongs_to_required_by_default = false`?

**Comment by Fudoshiki** [comment]
1. Do I need revert changes for .gitignore and rubocop style?
2. I trying add this option, can you help me

File: spec/app/models/author.rb [hunk 1]
```diff
@@ -1,6 +1,5 @@
 class Author
   include Mongoid::Document
   field :name, type: String
-
```
Comment (simi) [inline spec/app/models/author.rb]: :scissors:

File: README.md [hunk 2]
```diff
@@ -14,21 +14,6 @@ Add this line to your application's Gemfile:
 gem 'mongoid_paranoia'
 ```
 
-## Changes in 4.0
```
Comment (simi) [inline README.md]: Why removed?

**Comment by simi** [comment]
Very nice @Fudoshiki. Waiting for @johnnyshields to do final review also.

File: README.md (hunk 2 above)
Comment (Fudoshiki) [inline README.md]: Rails 5 deprecated validates_uniqueness_of syntax

Comment (simi) [inline README.md]: Hmm, do you have any link for that?

Anyway is 

```ruby
validates :title, :uniqueness => true
```

still valid?

If so, we need to keep this information and remove `validates_uniqueness_of` example only.

Comment (Fudoshiki) [inline README.md]: Will do

**Comment by simi** [comment]
Thanks. I've spotted beta was released today also.

@johnnyshields are you interested in this or should I handle this on my own?

**Comment by johnnyshields** [comment]
this looks fine, sorry I missed your earlier message. Thanks!

**Comment by Fudoshiki** [comment]
@simi, @johnnyshields  I don't worry, my projects works perfect with my branch, but this PR posted month ago, open source community e.t.c 😃

**Comment by Fudoshiki** [comment]
mongoid-slug related now on my repository, this wrong)

**Comment by Fudoshiki** [comment]
@johnnyshields @simi ?

**Comment by simi** [comment]
:balloon:  @Fudoshiki merged! :tada:

**Pull request merged**